internal/splunkconfig/config/indexarchiver.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import "fmt"
+
+// Archiver represents the value of an Index's "archiver.coldStorageProvider" fields.
+type Archiver string
```

Comment:
Any reason to not leverage a struct for the Archiver instead of string?

Reply:
In my mind, if you make it a struct with a few fields you can expand this in the future to other cloud provider cold storages...

Reply:
IndexArchiver is meant to be an enum, not an object, that is why I made it as a string.